src/main/java/io/confluent/connect/elasticsearch/ElasticsearchWriter.java

```diff
@@ -131,188 +93,102 @@
-     */
-    public Builder setMaxBufferedRecords(long maxBufferedRecords) {
+    public Builder setMaxBufferedRecords(int maxBufferedRecords) {
       this.maxBufferedRecords = maxBufferedRecords;
       return this;
     }
 
-    /**
-     * Set the max number of inflight requests.
-     * @param maxInFlightRequests The max allowed number of inflight requests.
-     * @return an instance of ElasticsearchWriter Builder.
-     */
     public Builder setMaxInFlightRequests(int maxInFlightRequests) {
       this.maxInFlightRequests = maxInFlightRequests;
       return this;
     }
 
-    /**
-     * Set the number of requests to process as a batch when writing.
-     * to Elasticsearch.
-     * @param batchSize the size of each batch.
-     * @return an instance of ElasticsearchWriter Builder.
-     */
     public Builder setBatchSize(int batchSize) {
       this.batchSize = batchSize;
       return this;
     }
 
-    /**
-     * Set the linger time.
-     * @param lingerMs The linger time to use in milliseconds.
-     * @return an instance of ElasticsearchWriter Builder.
-     */
     public Builder setLingerMs(long lingerMs) {
       this.lingerMs = lingerMs;
       return this;
     }
 
-    /**
-     * Set the max retry for a batch
-     * @param maxRetry The number of max retry.
-     * @return an instance of ElasticsearchWriter Builder.
-     */
     public Builder setMaxRetry(int maxRetry) {
       this.maxRetry = maxRetry;
       return this;
     }
 
-    /**
-     * Set the retry backoff.
-     * @param retryBackoffMs The retry backoff in milliseconds.
-     * @return an instance of ElasticsearchWriter Builder.
-     */
     public Builder setRetryBackoffMs(long retryBackoffMs) {
       this.retryBackoffMs = retryBackoffMs;
       return this;
     }
 
-    /**
-     * Set the SinkTaskContext
-     * @param context The SinkTaskContext.
-     * @return an instance of ElasticsearchWriter Builder.
-     */
-    public Builder setContext(SinkTaskContext context) {
-      this.context = context;
-      return this;
-    }
-
-    /**
-     * Build the ElasticsearchWriter.
-     * @return an instance of ElasticsearchWriter.
-     */
     public ElasticsearchWriter build() {
       return new ElasticsearchWriter(
-          client, type, ignoreKey, ignoreSchema, topicConfigs, flushTimeoutMs, maxBufferedRecords, maxInFlightRequests, batchSize, lingerMs, maxRetry, retryBackoffMs, context);
+          client, type, ignoreKey, ignoreSchema, topicConfigs, flushTimeoutMs, maxBufferedRecords, maxInFlightRequests, batchSize, lingerMs, maxRetry, retryBackoffMs
+      );
     }
   }
 
   public void write(Collection<SinkRecord> records) {
-    if (bulkProcessor.getException() != null) {
-      throw new ConnectException("BulkProcessor failed with non-retriable exception", bulkProcessor.getException());
-    }
-    if (bulkProcessor.getTotalBufferedRecords() + records.size() > maxBufferedRecords) {
```

Comment:
This represents a behavior change, now it's going to block in `bulkProcessor.add()` till buffer space becomes available. Maybe it's better to throw a `RetriableException` if it helps with consumer liveness? If so I can restore that.